gems/aws-sdk-s3/lib/aws-sdk-s3/encryptionV2/io_encrypter.rb

```diff
@@ -52,8 +52,12 @@ def encrypt_to_stringio(cipher, plain_text)
         def encrypt_to_tempfile(cipher, io)
           encrypted = Tempfile.new(self.object_id.to_s)
           encrypted.binmode
-          while chunk = io.read(ONE_MEGABYTE)
-            encrypted.write(cipher.update(chunk))
+          while chunk = io.read(ONE_MEGABYTE, read_buffer ||= String.new)
+            if cipher.method(:update).arity == 1
```

Comment:
Yuck...

Reply:
Agreed - I don't love this.  I don't think we can count on the versions of openssl a user has so we have three options:
1. Check openssl versions instead of arity.  This is less direct but maybe makes it more clear what why we're doing this.  
2.  Create a wrapper around cipher to normalize this interface difference (push this mess down).  
3. Drop this change -  >80% of our users would see improved performance from this.

I lean towards taking this change with the arity check (with better documentation in comments) since its a fairly significant improvement in performance for the bulk of our users. 

Side note: I created a seperate PR for updating the travis.yml: #2340